### PR TITLE
feat!: Drop Rails 5 Support

### PR DESCRIPTION
**Pull request by arielvalentin**
Closes https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/249

### CODE REVIEW

**Review by ericmustin — APPROVED** [review]

File: instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency 'opentelemetry-api', '~> 1.0'
   spec.add_dependency 'opentelemetry-instrumentation-base', '~> 0.21.0'
 
-  spec.add_development_dependency 'activejob', '>= 5.2.0'
+  spec.add_development_dependency 'activejob', '>= 6.0.0', '< 8.0.0'
```
Comment (ahayworth) [inline instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec:31]: Any reason not to optimistically support future activejobs?

**Review by ahayworth — APPROVED** [review]
Overall LGTM but I did have one question about the dependency version changes (but no real worries about it either way)

File: instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec (hunk 1 above)
Comment (arielvalentin) [reply instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec:31]: Personally speaking I think gem requirements send a more explicit signal about what we support and what we don't but I do not have to add that constraint here if you prefer I remove it.

Comment (ahayworth) [reply instrumentation/active_job/opentelemetry-instrumentation-active_job.gemspec:31]: 🤷 Honestly I don't have any strong feelings either way; I wasn't sure if it was an implicit signal that there may be problems with future versions or something. 😄

**Comment by arielvalentin** [comment]
Summarizing our discussion from SIG on 2023-01-07

Before Merging This PR...

1. Open a discussion and post in CNCF slack about this PR along with release timelines
2. Open a PR that prints deprecation warnings and release a patch version of the gem
3. Provide instructions on what version to pin OTel instrumentation for Rails 5 apps

**Comment by arielvalentin** [comment]
@ahayworth et.al. I have had yet _another_ change of heart. 

I would like to merge this before migrating active support instrumentation. I would like to minimize any challenges we will run into due to workarounds with limitations in Rails 5 when switching to AS. 

I will also remove the restrictions I added to development dependencies before merging

**Review by ahayworth — APPROVED** [review]
🚢

**Pull request merged**